environment/src/main/java/com/sequenceiq/environment/environment/flow/creation/handler/ResourceGroupCreationHandler.java

```diff
@@ -0,0 +1,168 @@
+
+    private ParametersService parameterService;
+
+    private Clock clock;
+
+    protected ResourceGroupCreationHandler(EventSender eventSender,
+            CloudPlatformConnectors cloudPlatformConnectors,
+            AzureClientService azureClientService,
+            CredentialToCloudCredentialConverter credentialToCloudCredentialConverter,
+            AzureSingleResourceGroupFeatureSwitch azureSingleResourceGroupFeatureSwitch,
+            CostTagging costTagging, EventBus eventBus, EnvironmentService environmentService,
+            ParametersService parameterService,
+            Clock clock) {
+        super(eventSender);
+        this.cloudPlatformConnectors = cloudPlatformConnectors;
+        this.azureClientService = azureClientService;
+        this.credentialToCloudCredentialConverter = credentialToCloudCredentialConverter;
+        this.azureSingleResourceGroupFeatureSwitch = azureSingleResourceGroupFeatureSwitch;
+        this.costTagging = costTagging;
+        this.eventBus = eventBus;
+        this.environmentService = environmentService;
+        this.parameterService = parameterService;
+        this.clock = clock;
+    }
+
+    @Override
+    public void accept(Event<EnvironmentDto> environmentDtoEvent) {
+        EnvironmentDto environmentDto = environmentDtoEvent.getData();
+        environmentService.findEnvironmentById(environmentDto.getId())
+                .ifPresentOrElse(environment -> {
+                            try {
+                                createResourceGroup(environmentDto, environment);
+                                goToNetworkCreationState(environmentDtoEvent);
+                            } catch (Exception e) {
+                                goToFailedState(environmentDtoEvent, e.getMessage());
+                            }
+                        }, () -> goToFailedState(environmentDtoEvent, String.format("Environment was not found with id '%s'.", environmentDto.getId()))
+                );
+    }
+
+    private void createResourceGroup(EnvironmentDto environmentDto, Environment environment) {
+        if (!azureSingleResourceGroupFeatureSwitch.isActive()) {
+            LOGGER.debug("Azure single resource group feature turned off, not creating resourcegroup.");
+            return;
+        }
+        Optional<AzureResourceGroupDto> azureResourceGroupDtoOptional = getAzureResourceGroupDto(environmentDto);
+        if (!AZURE.name().equals(environmentDto.getCloudPlatform()) || azureResourceGroupDtoOptional.isEmpty()) {
+            LOGGER.debug("Cloudplatform not azure or no azure resource group dto defined, not creating resource group.");
+            return;
+        }
+        AzureResourceGroupDto azureResourceGroupDto = azureResourceGroupDtoOptional.get();
+        if (!CREATE_NEW.equals(azureResourceGroupDto.getResourceGroupCreation())) {
+            LOGGER.debug("Azure resource group creation not requested, not creating resource group.");
```

Comment:
Azure resource group creation not requested -> New single Azure resource group creation not requested